kinto/core/utils.py

```diff
@@ -443,3 +444,40 @@ def instance_uri(request, resource_name, **params):
     """Return the URI for the given resource."""
     return strip_uri_prefix(request.route_path('%s-record' % resource_name,
                                                **params))
+
+
+def apply_json_patch(record, ops):
+    """
+    Apply JSON Patch operations using jsonpatch.
+
+    :param record: base record where changes should be applied (not in-place).
+    :param list changes: list of JSON patch operations.
+    :param bool only_data: param to limit the scope of the patch only to 'data'.
+    :returns dict data: patched record data.
+             dict permissions: patched record permissions
+    """
+    data = record.copy()
+    permissions = {'read': set(), 'write': set()}
+
+    # To be used with SharableResource object (patching permissions)
+    if '__permissions__' in record:
+        permissions.update(record['__permissions__'])
+
+    # Map permissions as dicts since jsonpatch module doesn't accept sets.
+    permissions = {k: {i: i for i in v} for k, v in permissions.items()}
+
+    resource = {'data': data, 'permissions': permissions}
+
+    # Allow patch permissions without value since key and value are equal on sets
+    for op in ops:
+        if 'path' in op:
+            if op['path'].startswith(('/permissions/read/',
+                                      '/permissions/write/')):
+                op['value'] = op['path'].split('/')[-1]
+
+    result = jsonpatch.apply_patch(resource, ops)
+
+    data = result['data']
+    permissions = result['permissions']
+
+    return data, permissions
```

Comment:
I think it's acceptable to return `result` here. We give an input and a patch, we receive an output. If you need to keep them separate, then you could also take two separate arguments as input